system-contracts/contracts/precompiles/Sekp256r1.yul

```diff
@@ -0,0 +1,108 @@
+            ////////////////////////////////////////////////////////////////
+
+            /// @notice Group order of secp256r1, see https://eips.ethereum.org/EIPS/eip-7212
+            function SECP256K1_GROUP_SIZE() -> ret {
+                ret := 0xffffffff00000000ffffffffffffffffbce6faada7179e84f3b9cac2fc632551
+            }
+
+            /// @notice Curve prime field modulus, see https://eips.ethereum.org/EIPS/eip-7212
+            function PRIME_FIELD_MODULUS() -> ret {
+                ret := 0xffffffff00000001000000000000000000000000ffffffffffffffffffffffff
+            }
+
+            /// @dev The gas cost of processing sekp256r1 circuit precompile.
+            function SEKP256_VERIFY_GAS_COST() -> ret {
+                ret := 12000
+            }
+
+            ////////////////////////////////////////////////////////////////
+            //                      HELPER FUNCTIONS
+            ////////////////////////////////////////////////////////////////
+            
+            /// @dev Packs precompile parameters into one word.
+            /// Note: functions expect to work with 32/64 bits unsigned integers.
+            /// Caller should ensure the type matching before!
+            function unsafePackPrecompileParams(
+                uint32_inputOffsetInWords,
+                uint32_inputLengthInWords,
+                uint32_outputOffsetInWords,
+                uint32_outputLengthInWords,
+                uint64_perPrecompileInterpreted
+            ) -> rawParams {
+                rawParams := uint32_inputOffsetInWords
+                rawParams := or(rawParams, shl(32, uint32_inputLengthInWords))
+                rawParams := or(rawParams, shl(64, uint32_outputOffsetInWords))
+                rawParams := or(rawParams, shl(96, uint32_outputLengthInWords))
+                rawParams := or(rawParams, shl(192, uint64_perPrecompileInterpreted))
+            }
+
+            /// @dev Executes the `precompileCall` opcode.
+            function precompileCall(precompileParams, gasToBurn) -> ret {
+                // Compiler simulation for calling `precompileCall` opcode
+                ret := verbatim_2i_1o("precompile", precompileParams, gasToBurn)
+            }
+
+            ////////////////////////////////////////////////////////////////
+            //                      FALLBACK
+            ////////////////////////////////////////////////////////////////
+
+            let digest := calldataload(0)
+            let r := calldataload(32)
+            let s := calldataload(64)
+            let x := calldataload(96)
+            let y := calldataload(128)
+
+            // Validate the input by the yellow paper rules (Appendix E. Precompiled contracts)
+            let sIsInvalid := or(eq(s, 0), gt(s, sub(SECP256K1_GROUP_SIZE(), 1)))
+            let rIsInvalid := or(eq(r, 0), gt(r, sub(SECP256K1_GROUP_SIZE(), 1)))
+
+            if or(sIsInvalid, rIsInvalid) {
+                return(0, 0)
+            }
```

Comment:
This is also checked in circuits, shall we remove the check here? @StanislavBreadless @shamatar ?

Reply:
Then you can even use fat pointer to propagate calldata to circuits

Reply:
Propagating calldata is too much work for a small gain for this precompile. It reads from heap because heap values are aligned

Reply:
And yes, "r" and "s" checks are present in the circuit